### PR TITLE
Add spring autoconfiguration for metrics and prometheus reporter

### PR DESCRIPTION
**Pull request by objectiser**
This PR contains code moved from this PR: https://github.com/opentracing-contrib/java-spring-cloud/pull/15

An example of how it simplifies an example can be seen in this PR: https://github.com/objectiser/opentracing-prometheus-example/pull/13

### CODE REVIEW

File: opentracing-metrics-prometheus-spring-autoconfigure/pom.xml [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+      <artifactId>opentracing-metrics-spring-autoconfigure</artifactId>
+      <version>${project.version}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>io.opentracing.contrib</groupId>
+      <artifactId>opentracing-metrics-prometheus</artifactId>
+      <version>${project.version}</version>
+      <exclusions>
+      	<exclusion>
```
Comment (jpkrohling) [inline opentracing-metrics-prometheus-spring-autoconfigure/pom.xml]: Spacing looks odd here

File: opentracing-metrics-spring-autoconfigure/pom.xml [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+      <version>${version.org.springframework.boot}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>io.opentracing.contrib</groupId>
+      <artifactId>opentracing-metrics-prometheus</artifactId>
+      <version>${project.version}</version>
+      <exclusions>
+      	<exclusion>
+     	  <groupId>io.prometheus</groupId>
```
Comment (jpkrohling) [inline opentracing-metrics-spring-autoconfigure/pom.xml]: Spacing again

**Review by jpkrohling — APPROVED** [review]
Apart from two spacing issues, LGTM (as usual, I can't judge the Spring semantics)

File: opentracing-metrics-prometheus-spring-autoconfigure/pom.xml (hunk 1 above)
Comment (pavolloffay) [reply opentracing-metrics-prometheus-spring-autoconfigure/pom.xml]: Why do you exclude it? And why are other promo deps provided? I think this should pull-in all what is needed.

File: opentracing-metrics-spring-autoconfigure/pom.xml [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+
+    <dependency>
+      <groupId>org.springframework.boot</groupId>
+      <artifactId>spring-boot-autoconfigure</artifactId>
+      <version>${version.org.springframework.boot}</version>
+    </dependency>
+
+    <dependency>
+      <groupId>io.opentracing.contrib</groupId>
+      <artifactId>opentracing-metrics-prometheus</artifactId>
```
Comment (pavolloffay) [inline opentracing-metrics-spring-autoconfigure/pom.xml:47]: What is the purpose of this artifact? There is one for prometheus so why this has some provided prometheus deps?

**Review by pavolloffay — APPROVED** [review]
dependencies look weird but maybe I am missing something.

File: opentracing-metrics-spring-autoconfigure/pom.xml (hunk 3 above)
Comment (objectiser) [reply opentracing-metrics-spring-autoconfigure/pom.xml:47]: Have removed the exclusions - it was left over from when in the java-spring-cloud repo, so the autoconfigs would become active if prometheus was in the classpath.

**Comment by objectiser** [comment]
@jpkrohling @pavolloffay Thanks!

**Pull request merged**